implementations/rust/ockam/ockam_command/src/lib.rs

```diff
@@ -447,3 +449,78 @@ pub(crate) fn replace_hyphen_with_stdin(s: String) -> String {
         s
     }
 }
+
+const ENV_FORCE_COLOR: &str = "ockam_force_color";
+
+fn show_help(help: clap::Error) {
+    use std::env;
+    let mut try_fallback = false;
+    let preferred_pager = env::var_os("PAGER").unwrap_or_else(|| {
+        try_fallback = true;
+        OsString::from("less")
+    });
+
+    if preferred_pager == "false" {
+        use clap::{ColorChoice::*, CommandFactory};
+        let possibly_forced = if env::var_os(ENV_FORCE_COLOR).is_some() {
+            Always
+        } else {
+            Auto
+        };
+
+        help.with_cmd(&OckamCommand::command().color(possibly_forced))
+            .exit();
+    }
+
+    if let Ok(()) = paginate_with(preferred_pager, &help) {
+        return;
+    }
+    if try_fallback {
+        if let Ok(()) = paginate_with(OsString::from("more"), &help) {
+            return;
+        }
+    }
+    paginate_with(OsString::from("false"), &help)
+        .expect("displaying help without pagination should always work");
+}
+
+fn paginate_with(pager: OsString, help: &clap::Error) -> Result<()> {
+    let mut pager_invocation = process::Command::new(&pager);
+    if Path::new(&pager)
+        .file_name()
+        .map_or("", |s| s.to_str().unwrap_or(""))
+        == "less"
```

Comment:
I'm not sure how portable would this be across different OS's. I still vote for using `which`. It will also look cleaner and it lets you check if the binary actually exists in the system.